### PR TITLE
Remove OAuth code specific for Desktop app

### PR DESCRIPTION
**Pull request by jsnajdr**
After the Desktop App build of Calypso was practically removed in #49614, including the `config/desktop.json` file, let's remove the desktop-specific OAuth code: the `/oauth-login` login page and the server-side `/oauth`, `/sms` and `/logout` endpoints that the app was using.

I also removed the `config/desktop-development.json` file, and removed a few instances of `config.isEnabled( 'desktop' )`, although there are still many remaining ones.

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
LGTM, great cleanup 👍 

> I also removed the `config/desktop-development.json` file, and removed a few instances of `config.isEnabled( 'desktop' )`, although there are still many remaining ones.

Yep, that would be a nice follow-up cleanup. I can see there are some fuzzy ones as well, like [this one](https://github.com/Automattic/wp-calypso/blob/remove/desktop-oauth/client/blocks/get-apps/index.jsx#L25).

File: client/auth/login.jsx [hunk 1]
```diff
@@ -1,214 +0,0 @@
...
-
-	render() {
-		const { translate } = this.props;
-		const { requires2fa, inProgress, errorMessage, errorLevel, showInstructions } = this.state;
-
-		return (
-			<Main className="auth">
-				<div className="auth__content">
-					<WordPressLogo />
-					<form className="auth__form" onSubmit={ this.submitForm }>
```
Comment (tyxla) [inline client/auth/login.jsx:132]: Just realized there are some related `z-index` definitions we could cleanup from [this file](https://github.com/Automattic/wp-calypso/blob/trunk/client/assets/stylesheets/shared/functions/_z-index.scss#L70).

Comment (jsnajdr) [reply client/auth/login.jsx:132]: Fixed in e062796.

**Comment by jsnajdr** [comment]
> I can see there are some fuzzy ones as well, like this one.

Yes, that's what discouraged me from removing them all now. Showing promo banner for the Desktop app is indeed undesired when you're already in Desktop app. We'll need to find some way to replace (or re-introduce) the `config.isEnabled( 'desktop' )` check with something else.

**Comment by belcherj** [comment]
This is great! Thank you.

**Comment by nsakaimbo** [comment]
Thanks for this cleanup, @jsnajdr! 

Yes, there are a few instances of `config.isEnabled( 'desktop' )` checks remaining, and generally speaking we should need far fewer of them than before. There are cases in which the check is still valid (such as showing the promo banner for the Desktop app as you mentioned), but generally speaking we should be able to prune most of them now.

FYI the check for `config.isEnabled( 'desktop' )` is now driven by the presence of the `window.electron` object ([link](https://github.com/Automattic/wp-calypso/pull/49614/files#diff-149f15d7e7207360a24544aed80518404c87168692e4c1aaf7cb0b96321ef211)). CC: @tyxla

**Comment by jsnajdr** [comment]
> the check for `config.isEnabled( 'desktop' )` is now driven by the presence of the `window.electron` object

I see, the `desktop` flag is still there and is set by a config override. Neat :+1:

**Pull request merged**